### ISSUE TITLE
Tests gradients by overfitting a single batch

### ISSUE THREAD

**Issue by Imartinezcuevas** [Imartinezcuevas/nano-llm#28]
Before training on the full dataset, we must ensure the model can learn.
